vendor/engines/secure_rooms/app/services/secure_rooms/access_manager.rb

```diff
@@ -0,0 +1,48 @@
+module SecureRooms
+
+  class AccessManager
+
+    attr_reader :verdict
+
+    def initialize(user, card_reader, requested_account_id)
+      @user = user
+      @card_reader = card_reader
+      @requested_account_id = requested_account_id
+    end
+
+    def process
+      @verdict = create_verdict
```

Comment:
This is just an idea: what if the verdict could house a selected account (for grants) and a list of accounts (for pending)? Then this method could return a verdict, so you wouldn't need to use the instance of this class anymore since there would be less state.